drivers/storage/portworx/component/plugin.go

```diff
@@ -0,0 +1,338 @@
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating %s config map  %s ", PluginConfigMapName, err)
+	}
+
+	// create console plugin
+	if err := p.createConsolePlugin(consolePluginFileName, cluster.Namespace, ownRef); err != nil {
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating console plugin %s ", err)
+	}
+
+	if len(errList) > 0 {
+		return commonerrors.New(strings.Join(errList, " , "))
+	}
+	return nil
+}
+
+func (p *plugin) Delete(cluster *corev1.StorageCluster) error {
+	var errList []string
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	// delete plugin configs
+	if err := k8s.DeleteDeployment(p.client, PluginDeploymentName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteConfigMap(p.client, PluginConfigMapName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteService(p.client, PluginServiceName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+
+	// delete nginx configs
+	if err := k8s.DeleteDeployment(p.client, NginxDeploymentName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteConfigMap(p.client, NginxConfigMapName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteService(p.client, NginxServiceName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+
+	// delete console plugin
+	if err := k8s.DeleteConsolePlugin(p.client, PluginName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if len(errList) > 0 {
+		return commonerrors.New(strings.Join(errList, " , "))
+	}
+	return nil
+}
+
+func (p *plugin) MarkDeleted() {
+	p.isOperatorCreated = false
+}
+
+// RegisterPortworxPluginComponent registers the PortworxPlugin component
+func RegisterPortworxPluginComponent() {
+	Register(PluginComponentName, &plugin{})
+}
+
+func init() {
+	RegisterPortworxPluginComponent()
+}
+
+func (p *plugin) createDeployment(filename, deploymentName string, ownerRef *metav1.OwnerReference, cluster *corev1.StorageCluster) error {
+	deployment, err := k8s.GetDeploymentFromFile(filename, pxutil.SpecsBaseDir())
+	if err != nil {
+		return err
+	}
+	deployment.Namespace = cluster.Namespace
+	deployment.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+
+	existingDeployment := &appsv1.Deployment{}
+	getErr := p.client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      deploymentName,
+			Namespace: cluster.Namespace,
+		},
+		existingDeployment,
+	)
+	if getErr != nil && !errors.IsNotFound(getErr) {
+		return getErr
+	}
+
+	equal, _ := util.DeepEqualPodTemplate(&deployment.Spec.Template, &existingDeployment.Spec.Template)
```

Comment:
We should be doing a comparison after we apply the below settings to the deployment. Otherwise it will always show up as different.